Still seeing peer certificate errors in the log (and thus a failure to sign in / use native network-dependent features):

```
Caused by:
    0: error sending request for url (https://api.github.com/repos/zed-industries/copilot/releases/tags/v0.7.0)
    1: client error (Connect)
    2: invalid peer certificate: BadSignature
2025-02-11T11:26:41.238868-05:00 [INFO] set environment variables from shell:/bin/zsh, path:/Users/matt/.sst/bin:/Users/matt/.rbenv/shims:/Users/matt/.nvm/versions/node/v21.5.0/bin:/Users/matt/repos/google-cloud-sdk/bin:/usr/local/opt/curl/bin:/usr/local/bin:/usr/local/sbin:/Users/matt/repos/go/bin:/Users/matt/.cargo/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Users/matt/.local/bin
2025-02-11T11:26:53.798608-05:00 [INFO] set status on client 253412: Reauthenticating
2025-02-11T11:26:53.811664-05:00 [INFO] set status on client 253412: Reconnecting
2025-02-11T11:26:53.871206-05:00 [INFO] set status on client 253412: ConnectionError
2025-02-11T11:26:53.871445-05:00 [ERROR] error sending request for url (https://zed.dev/rpc)

Caused by:
    0: error sending request for url (https://zed.dev/rpc)
    1: client error (Connect)
    2: invalid peer certificate: BadSignature
```

... and the expected logspam:

```
2025-02-11T11:16:08.751689-05:00 [INFO] set status on client 253412: Reauthenticating
2025-02-11T11:16:08.752258-05:00 [INFO] set status on client 253412: Reconnecting
2025-02-11T11:16:08.866154-05:00 [INFO] set status on client 253412: ConnectionError
2025-02-11T11:16:08.866458-05:00 [ERROR] failed to connect error sending request for url (https://zed.dev/rpc?preview=1)
2025-02-11T11:16:08.866668-05:00 [INFO] set status on client 253412: ReconnectionError { next_reconnection: Instant { tv_sec: 4895, tv_nsec: 421187333 } }
2025-02-11T11:16:18.902-05:00 [INFO] set status on client 253412: Reauthenticating
2025-02-11T11:16:18.902933-05:00 [INFO] set status on client 253412: Reconnecting
2025-02-11T11:16:18.966199-05:00 [INFO] set status on client 253412: ConnectionError
2025-02-11T11:16:18.966561-05:00 [ERROR] failed to connect error sending request for url (https://zed.dev/rpc?preview=1)
2025-02-11T11:16:18.966914-05:00 [INFO] set status on client 253412: ReconnectionError { next_reconnection: Instant { tv_sec: 4905, tv_nsec: 521544458 } }
2025-02-11T11:16:29.013301-05:00 [INFO] set status on client 253412: Reauthenticating
2025-02-11T11:16:29.013906-05:00 [INFO] set status on client 253412: Reconnecting
2025-02-11T11:16:29.099209-05:00 [INFO] set status on client 253412: ConnectionError
2025-02-11T11:16:29.09963-05:00 [ERROR] failed to connect error sending request for url (https://zed.dev/rpc?preview=1)
2025-02-11T11:16:29.099994-05:00 [INFO] set status on client 253412: ReconnectionError { next_reconnection: Instant { tv_sec: 4915, tv_nsec: 654736875 } }
```

--- a/crates/client/Cargo.toml
+++ b/crates/client/Cargo.toml
@@ -33,7 +33,6 @@ postage.workspace = true
 rand.workspace = true
 release_channel.workspace = true
 rpc = { workspace = true, features = ["gpui"] }
-rustls-platform-verifier.workspace = true
 rustls.workspace = true
 schemars.workspace = true
 serde.workspace = true

--- a/crates/client/Cargo.toml
+++ b/crates/client/Cargo.toml
@@ -33,7 +33,6 @@ postage.workspace = true
 rand.workspace = true
 release_channel.workspace = true
 rpc = { workspace = true, features = ["gpui"] }
-rustls.workspace = true
 schemars.workspace = true
 serde.workspace = true
 serde_json.workspace = true

--- a/crates/client/src/client.rs
+++ b/crates/client/src/client.rs
@@ -146,8 +146,6 @@ pub fn init_settings(cx: &mut App) {
 }
 
 pub fn init(client: &Arc<Client>, cx: &mut App) {
-    let _ = rustls::crypto::aws_lc_rs::default_provider().install_default();
-
     let client = Arc::downgrade(client);
     cx.on_action({
         let client = client.clone();

--- a/crates/client/src/client.rs
+++ b/crates/client/src/client.rs
@@ -26,8 +26,6 @@ use postage::watch;
 use rand::prelude::*;
 use release_channel::{AppVersion, ReleaseChannel};
 use rpc::proto::{AnyTypedEnvelope, EnvelopedMessage, PeerId, RequestMessage};
-use rustls::ClientConfig;
-use rustls_platform_verifier::ConfigVerifierExt;
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 use settings::{Settings, SettingsSources};
@@ -1128,13 +1126,11 @@ impl Client {
 
             match url_scheme {
                 Https => {
-                    let client_config = ClientConfig::with_platform_verifier();
-
                     let (stream, _) =
                         async_tungstenite::async_tls::client_async_tls_with_connector(
                             request,
                             stream,
-                            Some(client_config.into()),
+                            Some(http_client::tls_config().into()),
                         )
                         .await?;
                     Ok(Connection::new(

--- a/crates/http_client/Cargo.toml
+++ b/crates/http_client/Cargo.toml
@@ -25,3 +25,5 @@ log.workspace = true
 serde.workspace = true
 serde_json.workspace = true
 url.workspace = true
+rustls.workspace = true
+rustls-platform-verifier.workspace = true

--- a/crates/http_client/src/http_client.rs
+++ b/crates/http_client/src/http_client.rs
@@ -22,7 +22,16 @@ static TLS_CONFIG: OnceLock<rustls::ClientConfig> = OnceLock::new();
 
 pub fn tls_config() -> ClientConfig {
     TLS_CONFIG
-        .get_or_init(|| ClientConfig::with_platform_verifier())
+        .get_or_init(|| {
+            // rustls uses the `aws_lc_rs` provider by default
+            // This only errors if the default provider has already
+            // been installed. We can ignore this `Result`.
+            rustls::crypto::aws_lc_rs::default_provider()
+                .install_default()
+                .ok();
+
+            ClientConfig::with_platform_verifier()
+        })
         .clone()
 }
 

--- a/crates/http_client/src/http_client.rs
+++ b/crates/http_client/src/http_client.rs
@@ -8,13 +8,23 @@ pub use http::{self, Method, Request, Response, StatusCode, Uri};
 
 use futures::future::BoxFuture;
 use http::request::Builder;
+use rustls::ClientConfig;
+use rustls_platform_verifier::ConfigVerifierExt;
 #[cfg(feature = "test-support")]
 use std::fmt;
 use std::{
     any::type_name,
-    sync::{Arc, Mutex},
+    sync::{Arc, Mutex, OnceLock},
 };
 pub use url::Url;
+
+static TLS_CONFIG: OnceLock<rustls::ClientConfig> = OnceLock::new();
+
+pub fn tls_config() -> ClientConfig {
+    TLS_CONFIG
+        .get_or_init(|| ClientConfig::with_platform_verifier())
+        .clone()
+}
 
 #[derive(Default, Debug, Clone, PartialEq, Eq, Hash)]
 pub enum RedirectPolicy {

--- a/crates/reqwest_client/src/reqwest_client.rs
+++ b/crates/reqwest_client/src/reqwest_client.rs
@@ -51,7 +51,10 @@ impl ReqwestClient {
         }) {
             client = client.proxy(proxy);
         }
-        let client = client.build()?;
+
+        let client = client
+            .use_preconfigured_tls(http_client::tls_config())
+            .build()?;
         let mut client: ReqwestClient = client.into();
         client.proxy = proxy;
         Ok(client)

